### PR TITLE
Present forgot password errors to the user

### PR DESCRIPTION
**Pull request by arvindkrishnakumar-okta**
After this change, we would be able to propagate the below backend error to the UI for Forgot password case.

Backend error:

```
2021-05-02 22:35:46.681 ERROR 26698 --- [nio-8080-exec-8] c.o.i.s.a.c.IDXAuthenticationWrapper     : Exception occurred

com.okta.idx.sdk.api.exception.ProcessingException: Request to https://dev-30426356.okta.com/idp/idx/recover failed. HTTP status: 400
        at com.okta.idx.sdk.api.client.BaseIDXClient.handleErrorResponse(BaseIDXClient.java:505) ~[okta-idx-java-api-0.1.0-beta.6-SNAPSHOT.jar:0.1.0-beta.6-SNAPSHOT]
        at com.okta.idx.sdk.api.client.BaseIDXClient.recover(BaseIDXClient.java:413) ~[okta-idx-java-api-0.1.0-beta.6-SNAPSHOT.jar:0.1.0-beta.6-SNAPSHOT]
        at com.okta.idx.sdk.api.model.Recover.proceed(Recover.java:72) ~[okta-idx-java-api-0.1.0-beta.6-SNAPSHOT.jar:0.1.0-beta.6-SNAPSHOT]
        at com.okta.idx.sdk.api.client.IDXAuthenticationWrapper.populateForgotPasswordAuthenticatorUIOptions(IDXAuthenticationWrapper.java:387) ~[okta-idx-java-api-0.1.0-beta.6-SNAPSHOT.jar:0.1.0-beta.6-SNAPSHOT]
        at com.okta.spring.example.controllers.LoginController.handleForgotPassword(LoginController.java:135) ~[classes/:na]

2021-05-02 22:35:46.693 ERROR 26698 --- [nio-8080-exec-8] c.o.i.s.a.c.IDXAuthenticationWrapper     : Error Detail: [Reset password is not allowed at this time. Please contact support for assistance.]

```

image

image

### CODE REVIEW

File: api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java [hunk 1]
```diff
@@ -369,11 +369,10 @@ public AuthenticationResponse recoverPassword(String username) {
      * @return the list of AuthenticatorUIOptions
      */
     public List<AuthenticatorUIOption> populateForgotPasswordAuthenticatorUIOptions(
-            IDXClientContext idxClientContext) {
+            IDXClientContext idxClientContext) throws ProcessingException {
```
Comment (JayNewstrom) [inline api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java:372]: We haven't been exposing this in other methods. I wonder if it's best to return a different object here. One that contains a list of errors, and the list of UI options.

Or maybe attach some extra data to our `AuthenticationResponse` object.

**Review by JayNewstrom — APPROVED** [review]

File: api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java (hunk 1 above)
Comment (arvindkrishnakumar-okta) [reply api/src/main/java/com/okta/idx/sdk/api/client/IDXAuthenticationWrapper.java]: addressed this.

**Review by JayNewstrom — APPROVED** [review]
Want to do this in the other populateUIOptions method too? Can be a follow up, or I can do it too.

**Comment by arvindkrishnakumar-okta** [comment]
> Want to do this in the other populateUIOptions method too? Can be a follow up, or I can do it too.

I'll address that as well.

**Review by JayNewstrom — APPROVED** [review]

**Pull request merged**